src/Pos/Launcher/Runner.hs

```diff
@@ -185,15 +193,38 @@ runRawRealMode res np@NodeParams {..} sscnp listeners outSpecs (ActionSpec actio
        runDBHolder modernDBs . runCH np initNC $ initNodeDBs
        initTip <- runDBHolder modernDBs getTip
        stateM <- liftIO SM.newIO
+       stateM_ <- liftIO SM.newIO
+
+       -- TODO need an effect-free way of running this into IO.
+       let runIO :: forall t . RawRealMode ssc t -> IO t
+           runIO = runProduction .
```

Comment:
Everything looks ok.
Regarding described problem: yes, I see. I think we will do this refactoring after we unite all these holders into one holder (which is another refactoring).